Concepts/introspect.cc

```diff
@@ -15,8 +15,16 @@ struct has_sort_mem_fn : std::false_type {};
 template <typename T>
 struct has_sort_mem_fn<T, decltype((void)std::declval<T>().sort())> : std::true_type {};
 
+template <typename container, typename = void>
+void introspectingSort(container& cont) {
+	std::sort(std::begin(cont), std::end(cont));
+}
 
-
+template <typename container,
+          requires_(has_sort_mem_fn<container>())>
+void introspectingSort(container& cont) {
+	cont.sort();
+}
```

Comment:
If you enable/disable multiple instantiations via sfinae you have to make sure they're mutually exclusive otherwise you get ambiguities. What I mean by this is (pseudo code):

    enable if has .sort
    void sort();

    enable if !has .sort
    void sort();